### DISCUSSION TITLE
NHDS_Roots

### DISCUSSION THREAD

**Discussion by MSAfify** [danielver02/ALPS#167] [Q&A]
Hi,
I started using NHDS to get a quick solution before using ALPS, as mentioned in the ALPS paper. However, it seems difficult to generate the first and second roots in the NHDS paper "Daniel Verscharen and Benjamin D. G. Chandran 2018 Res. Notes AAS 2 13" in case of parallel k, considering the initial guess of frequency (imaginary) is (1.d-2, 1.d-12) and krange=0.01d0,2.d0. Could you please provide me with the initial guess for the first and second roots and the range of k? Also, please let me know which parameters in the NHDS code are important to distinguish between roots.

Please let me know how I can use ALPS to get electrostatic solutions.

Many thanks in advance!
Mahmoud

**Comment by kgklein** [comment]
Mahmoud- Have you tried running a map scan to find these roots? Fig. 2 from the 2018 ALPS publication shows the seven MHD scale solutions for kperp dp=kpar dp = 1E-3. Repeating that same calculation for the wavevectors of interest should help to build your intuition about how these modes change with wavevector, and provide you with the correct input values for the fast and Alfven solutions (as well as the slow and entropy modes).

To help speed up the ALPS frequency map calculations while you are considering just the bi-Maxwellian cases, you can turn both the protons and electrons to bimaxwellian mode (see the 'test_bimax.in' file for an example of turning the protons to using the NHDS solver built into ALPS via the 'use_bM' flag; see also the bM_spec_j namelist information in the input section of the readme page).

Cheers
Kris

**Reply by MSAfify** [reply]
Thanks!
